### PR TITLE
feat: play search result summary

### LINKED ISSUE #1325: ✨ [Feature request]: Add play search result summary
### Is your feature request related to a problem? Please describe.

Currently, when we search play(s), it just shows the resulting plays.

### Describe the solution you'd like.

We can add a summary of how many plays match the searched term similar to in GitHub when we search repositories.

Screenshot 2023-10-12 at 12 45 25 PM

Let's say we searched for "calcu" term. It should show "4 results for plays matching **calcu** sorted by Newest" below the search bar.

Screenshot 2023-10-12 at 12 55 23 PM

### Describe alternatives you've considered.

None

### Screenshot / Screenshare

_No response_

### Additional context

_No response_

### Record

- [X] I have checked the existing [issues](https://github.com/reactplay/react-play/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/reactplay/react-play/blob/main/CONTRIBUTING.md)
- [X] I want to work on this issue
- [X] I am in Hacktoberfest

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by janvi01**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Added Search Result Summary.
For example - we searched for "calcu" term. It should show "4 results for plays matching calcu sorted by Newest" below the search bar.

Fixes #1325

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I tested it locally on my development server.

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented on my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output
Screenshot 2023-10-17 at 1 33 39 PM

### CODE REVIEW

**Review by priyankarpal — CHANGES_REQUESTED** [review]
there are issues with responsiveness 



image

**Comment by janvi01** [comment]
@priyankarpal fixed it

**Comment by janvi01** [comment]
@atapas Kindly review.

File: src/common/playlists/PlayList.jsx [hunk 1]
```diff
@@ -62,6 +63,9 @@ const PlayList = () => {
         setIsFiltered(true);
       }
       setLoading(false);
+
+      // gets Search Term from url to display in the search summary
+      setGetSearchTerm(new URLSearchParams(location.search).get('text'));
```
Comment (Sachin-chaurasiya) [inline src/common/playlists/PlayList.jsx:68]: Can we use `useSearchParams` here provided by react-router?

File: src/common/playlists/PlayList.jsx [hunk 2]
```diff
@@ -18,6 +18,7 @@ const PlayList = () => {
   const [plays, setPlays] = useState();
   const [isFiltered, setIsFiltered] = useState(false);
   const [sortBy, setSortBy] = useState(localStorage.getItem('sortByPlay') || 'Newest');
+  const [getSearchTerm, setGetSearchTerm] = useState();
```
Comment (Sachin-chaurasiya) [inline src/common/playlists/PlayList.jsx:21]:
```suggestion
  const [searchTerm, setSearchTerm] = useState();
```

File: src/common/playlists/PlayList.jsx [hunk 3]
```diff
@@ -93,6 +97,14 @@ const PlayList = () => {
   return (
     <Fragment>
       {isFiltered ? null : <DynamicBanner randomPlay={randomPlay} />}
+      {location.search ? (
```
Comment (Sachin-chaurasiya) [inline src/common/playlists/PlayList.jsx:100]:
```suggestion
      {searchTerm ? (
```

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

**Review by Sachin-chaurasiya — APPROVED** [review]

**Comment by Sachin-chaurasiya** [comment]
@priyankarpal , please do the final review and merge it.

**Review by priyankarpal — APPROVED** [review]

**Pull request merged**